### DISCUSSION TITLE
"import net.minecraft.client.option.KeyBinding" was wrong

### DISCUSSION THREAD

**Discussion by ZhillerDev** [FabricMC/fabric-api#3251] [Mod Dev Support] [answered]
### Description

Hi!
Unfortunately, this issue has been bothering me for a few days now.
When I tried to use KeyBinding, I was prompted that I couldn't find this library.

Here is the code of import KeyBinding
`import net.minecraft.client.option.KeyBinding;`  

When I found the location of the library, I found that there was only one `ChatVisibility.java` file in the option folder, and there were no files related to KeyBinding

Perhaps I should download a new library and move it in? But I can't find anywhere to download.

**Comment by ZhillerDev** [comment, marked as answer]
problem solved

Just copy all the classes under the corresponding option in the `minecraft-clientonly-project-root-1.19-net.fabricmc.yarn.1_19.1.19+build.4-v2` package to `minecraft-common-project-root-1.19-net.fabricmc.yarn.1_19.1.19+build.4-v2` 

The package called by import during development is `minecraft-common` instead of `minecraft-clientonly`

**Comment by Daomephsta** [comment]
Fabric by default splits client-only Minecraft classes into a separate JAR that can only be accessed from `src/client`.
All code that needs to access client-only Minecraft classes must be in `src/client`. 
This catches uses of client-only classes on the server at compile time, rather than the mod crashing unexpectedly at runtime when installed on a dedicated server.

Copying the client-only classes into the common JAR defeats the purpose of split sources. I do not recommend it.
